src/remote.rs

```diff
@@ -238,38 +240,89 @@ impl Remote {
             target.pop();
 
             let url = format!("https://{}:{}/.well-known/jmap", target, name.port());
-            match http_wrapper.get_session(url.as_str()) {
-                Ok((session_url, session)) => {
-                    return Ok(Remote {
-                        http_wrapper,
-                        session_url,
-                        session,
-                    })
-                }
-
-                Err(e) => last_err = Some((url, e)),
+            match Self::open_url(url.as_str(), username, password, timeout) {
+                Ok(s) => return Ok(s),
+                Err(e) => last_err = Some(e),
             };
         }
         // All of them failed! Return the last error.
-        let (session_url, error) = last_err.unwrap();
-        Err(error).context(OpenSessionSnafu { session_url })
+        Err(last_err.unwrap())
```

Comment:
Hmm, it may be possible for `last_err` to be `None` here if we got a response with 0 entires. I don't really know what would cause that, but we should be cautious and add a new snafu error for that case.

Reply:
Yeah, I noticed that too, I just kept the existing behaviour (unless I read it wrong). I'll add something.

Reply:
From testing, it looks like `srv_lookup` fails if the query returns zero answers (`ResolveErrorKind::NoRecordsFound`), which fires `SrvLookupSnafu`. So the only way we get here is if the lookup succeeded, but none of the results led to a valid JMAP session. That should mean we can't even get here; there will _always_ be a last error.